pub/handler.go

```diff
@@ -0,0 +1,65 @@
+package pub
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/Iknite-Space/psss/models"
+	"github.com/rs/zerolog"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/sns"
+)
+
+type SNSPublisher struct {
+	SnsClient *sns.Client
+	logger    zerolog.Logger
+}
```

Comment:
I think it's be cleaner if the topic url was a field on this struct. That way publisher don't need to know anything about the underlying implementation.
```suggestion
type SNSPublisher[T proto.Message] struct {
	SnsClient *sns.Client
	topicArn string
	logger    zerolog.Logger
}
```

Reply:
Another way to think about this is, what if we changed the underlying implementation and instead of using SNS+SQS, we used RabbitMQ or Kafka? This interface wouldn't make sense anymore because these brokers wouldn't have any "topic urls"